include/sys/util.h

```diff
@@ -775,6 +866,127 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
  */
 #define UTIL_LISTIFY(LEN, F, ...) UTIL_EVAL(UTIL_REPEAT(LEN, F, __VA_ARGS__))
 
+/* Set of internal macros used for FOR_EACH series of macros. */
+#define Z_FOR_EACH_IDX(count, n, macro, semicolon, fixed_arg0, fixed_arg1, ...)\
+	UTIL_WHEN(count)						\
+	(								\
+		UTIL_OBSTRUCT(macro)					\
+		(							\
+			fixed_arg0, fixed_arg1, n, GET_ARG1(__VA_ARGS__)\
+		)semicolon						\
+		UTIL_OBSTRUCT(Z_FOR_EACH_IDX_INDIRECT) ()		\
+		(							\
+			UTIL_DEC(count), UTIL_INC(n), macro, semicolon, \
+			fixed_arg0, fixed_arg1,				\
+			GET_ARGS_LESS_1(__VA_ARGS__)			\
+		)							\
+	)
+
+#define Z_FOR_EACH_IDX_INDIRECT() Z_FOR_EACH_IDX
+
+#define Z_FOR_EACH_IDX2(count, iter, macro, sc, fixed_arg0, fixed_arg1, ...) \
+	UTIL_EVAL(Z_FOR_EACH_IDX(count, iter, macro, sc,\
+				 fixed_arg0, fixed_arg1, __VA_ARGS__))
+
+#define Z_FOR_EACH_SWALLOW_NOTHING(F, fixed_arg, index, arg) \
+	F(index, arg, fixed_arg)
+
+#define Z_FOR_EACH_SWALLOW_FIXED_ARG(F, fixed_arg, index, arg) F(index, arg)
+
+#define Z_FOR_EACH_SWALLOW_INDEX_FIXED_ARG(F, fixed_arg, index, arg) F(arg)
+#define Z_FOR_EACH_SWALLOW_INDEX(F, fixed_arg, index, arg) F(arg, fixed_arg)
+
+/**
+ * @brief Calls macro F for each provided argument with index as first argument
+ *	  and nth parameter as the second argument.
+ *
+ * Example:
+ *   \#define F(idx, x) int a\#\#idx = x;
+ *   FOR_EACH_IDX(F, 4, 5, 6)
```

Comment:
Very sorry for the late review.

This style of example is done in a few places in this file, and it doesn't render well:

![Screenshot from 2020-05-05 11-44-18](https://user-images.githubusercontent.com/62298616/81103824-740e8980-8ec6-11ea-94fe-7ddf11d8d19c.png)

You can see this for yourself in the `make doxygen` output in `doc/_build/doxygen/html/util_8h.html` for this macro. (This file isn't pulled in via Sphinx and Doxygen output is unofficial, but still: if this example is only meant to be read only in source code form, then the `\#\#` might as well be `##`, making it easy to read in that form.)

I think I was saying similar things in the original review to @de-nordic's related PRs to this file. To fix, extra newlines and at least four spaces for code blocks are needed:

```
 * Example:
 *
 *     #define F(idx, x) int a##idx = x;
 *     FOR_EACH_IDX(F, 4, 5, 6)
 *
 * will result in following code:
 *
 *     int a0 = 4;
 *     int a1 = 5;
 *     int a2 = 6;
```

This style is more readable in both source code form (no `\#\#`, so it's easy to copy/paste) and rendered doxygen also:


![Screenshot from 2020-05-05 11-52-18](https://user-images.githubusercontent.com/62298616/81104124-e5e6d300-8ec6-11ea-8c0d-2a6e4338922d.png)

Would you mind fixing these in the new code being added by this PR?  It's an issue for FOR_EACH_IDX_FIXED_ARG also and in another couple of places.